python_modules/dagster/dagster_tests/general_tests/test_internal_init_implementations.py

```diff
@@ -0,0 +1,33 @@
+import inspect
+from typing import Type
+
+import pytest
+from dagster._utils import IHasInternalInit
+
+
+@pytest.mark.parametrize("cls", IHasInternalInit.__subclasses__())
```

Comment:
Use of `__subclasses__` highly suspect. I didn't know that this even existed, but almost certainly such a construct is highly dependent on import order. A quick test bears this out:

```
class ISomeInterface:
    pass

print(ISomeInterface.__subclasses__())

class Foo(ISomeInterface):
    pass

print(ISomeInterface.__subclasses__())
```

```
(dagster-x86-3.10.3) elementl-13in-m1-mbppro-2021:dagster schrockn$ python subclasses.py
[]
[<class '__main__.Foo'>]
```

The right approach here is to either (a) explicitly whitelist definition classes or (b) import from top-level dagster and check all imports to see if they subclass from IHasInternalInit.

Reply:
I'll go ahead an add a top-level `import dagster as dagster` import. I think that should ensure that all user-exposed/top-level classes are loaded, e.g. equivalent to (b)?

Reply:
I feel pretty strongly that we shouldn't use something as fragile as `__subclasses__`. What if some lazy initialization is introduced or something?

Let's directly iterate over the top-level symbols in dagster and test for `isinstance`